### ISSUE TITLE
Ověření e mailové adresy v reálném čase

### ISSUE THREAD

**Issue by lukasprelovsky** [pavelevap/ceske-sluzby#45]
Zákazníci na woocommerce často zadávají špatný email, např namísto gmail.com zadají .cz a email jim nedorazí.

existuje řešení https://wordpress.org/plugins/woocommerce-email-validation/

toto řešení ale jen zdvojuje políčko emailu, aby ho zákazník napsal 2x.

napadla mě možnost přidat vertifikaci emailu v reálném čase, nic jsem už udělané na Woo ani nenašel.

Příklad ověření validity zde: http://email-checker.net/

**Comment by lukasprelovsky** [comment]
hlavní důvod, proč mi vadi že když zakaznik vyplnit zly email, tak se neodešle heureka dotaznik na Ověřeno zákazníky. Eshop uz nema moznost hodnotit přes Heureku. Takto přichází o recenze. To je hlavní důvod proč je nutně vymyslet způsob aby se nestalo ze zakaznici zadávat zly email :)

**Comment by pavelevap** [comment]
WooCommerce asi kontroluje správnost emailů po technické stránce, tedy `neco@domena.koncovka`. Pokud ale někdo zadá `neco@seynam.cz` nebo `neco@gmail.cz`, tak to asi neohlídá. Jaké jsou ještě nejčastější chyby?

Ta odkazovaná služba je patrně založená na tomto skriptu: http://php.vrana.cz/kontrola-e-mailove-adresy.php

Ale mohl by tam fungovat třeba tento plugin: https://wordpress.org/plugins/mailgun-email-validator/

**Comment by lukasprelovsky** [comment]
Ano, Woo nebo WP kontroluje vzdy jenom spravnost. Je mu jedno jestli je spatne napsana koncovka, nebo i neexistujuci koncovka napr.xyz ... 

Nejcastejsi chyby jsou takove , ze zadaji neplatni / nefungujici email.

Napr neco@gmail.sk namisto gmail.com .. 

Ta sluzba co jsem psal tak ta overuje primo jestli email funguje, tedy je aktivny. 

mailgun-email-validator bude fungovat asi jenom pro registrace, ne pro woocommerce ceckout v real time ..

**Comment by lukasprelovsky** [comment]
Ten mailgun je nejaka silena sluzba, ktera chce prepojeni s ich api, navic overeni domeny pres MX zaznamy :D

**Comment by pavelevap** [comment]
Ano, ta služba kontroluje doménu a MX záznamy (stejně jako odkazovaný skript a plugin). Ono není bohužel moc reálných způsobů, jak to zkontrolovat :-) Můžeme zkusit sérii opravných pravidel (ale tam je dost kombinací) nebo kontrolovat existenci domény a MX záznamu (a zkoušet doručitelnost).

Odkazovaný plugin by mohl fungovat i pro checkout, WooCommerce používá myslím také funkci `is_email()`, ale škoda, že je potřeba nějaký API klíč.

**Comment by lukasprelovsky** [comment]
Nevim overovat mx zaznamy domeny, propojovat s api, nastavovat zaznamy domeny i pro email, nevim zda se mi to hodne moc zbytecnosti. 

Tady http://mailgun.github.io/validator-demo/ to funguje uplne perfektne, kdyz zadas email ktery aktivny proste neni tak to ukaze chybu.

dela to tam vice overeni https://github.com/mailgun/validator-demo/tree/gh-pages
